diff --git a/node_modules/@strapi/plugin-users-permissions/server/controllers/user.js b/node_modules/@strapi/plugin-users-permissions/server/controllers/user.js
index f096bd8..a44c5eb 100644
--- a/node_modules/@strapi/plugin-users-permissions/server/controllers/user.js
+++ b/node_modules/@strapi/plugin-users-permissions/server/controllers/user.js
@@ -11,6 +11,8 @@ const utils = require('@strapi/utils');
 const { getService } = require('../utils');
 const { validateCreateUserBody, validateUpdateUserBody } = require('./validation/user');
 
+
+
 const { sanitize } = utils;
 const { ApplicationError, ValidationError, NotFoundError } = utils.errors;
 
@@ -33,7 +35,7 @@ module.exports = {
 
     await validateCreateUserBody(ctx.request.body);
 
-    const { email, username, role } = ctx.request.body;
+    const { email, username, role, toptalAlias } = ctx.request.body;
 
     const userWithSameUsername = await strapi
       .query('plugin::users-permissions.user')
@@ -69,6 +71,37 @@ module.exports = {
 
     try {
       const data = await getService('user').add(user);
+      const skillService = await getService('api::skill.skill');
+
+      console.log(data);
+      const skillsData = await Promise.all(_.uniq(skills).
+        map(async name => {
+        const existingSkill = await strapi.db.query('api::skill.skill')
+          .findOne({ // uid syntax: 'api::api-name.content-type-name'
+            where: {
+              name,
+            },
+            populate: {
+              users: true,
+            },
+        });
+
+        let skillBody;
+
+        if (existingSkill) {
+          skillBody = existingSkill;
+          skillBody.users.push(data);
+          return getService('api::skill.skill').update(skillBody);
+        }
+
+        skillBody = {
+          name,
+          users: [data]
+        };
+
+        return getService('api::skill.skill').create(skillBody);
+      }));
+
       const sanitizedData = await sanitizeOutput(data, ctx);
 
       ctx.created(sanitizedData);
diff --git a/node_modules/@strapi/plugin-users-permissions/server/services/providers-registry.js b/node_modules/@strapi/plugin-users-permissions/server/services/providers-registry.js
index a6be6d5..b0aa697 100644
--- a/node_modules/@strapi/plugin-users-permissions/server/services/providers-registry.js
+++ b/node_modules/@strapi/plugin-users-permissions/server/services/providers-registry.js
@@ -50,14 +50,18 @@ const getInitialProviders = ({ purest }) => ({
     const google = purest({ provider: 'google' });
 
     return google
-      .query('oauth')
-      .get('tokeninfo')
-      .qs({ accessToken })
+      .get('oauth2/v2/userinfo')
+      .qs({ access_token: accessToken })
       .request()
-      .then(({ body }) => ({
-        username: body.email.split('@')[0],
-        email: body.email,
-      }));
+      .then(({body}) => {
+        return ({
+          username: body.email.split('@')[0],
+          email: body.email,
+          name: body.name,
+          picture: body.picture,
+          toptalAlias: body.email.split('@')[0].replace('.', '-')
+        })
+      }).catch(e => console.log(e));
   },
   async github({ accessToken }) {
     const github = purest({
diff --git a/node_modules/@strapi/plugin-users-permissions/server/services/providers.js b/node_modules/@strapi/plugin-users-permissions/server/services/providers.js
index 79805b3..6af8175 100644
--- a/node_modules/@strapi/plugin-users-permissions/server/services/providers.js
+++ b/node_modules/@strapi/plugin-users-permissions/server/services/providers.js
@@ -7,6 +7,9 @@
 // Public node modules.
 const _ = require('lodash');
 const urlJoin = require('url-join');
+const scrape = require('website-scraper');
+const rimraf = require("rimraf");
+const HTMLParser = require('node-html-parser');
 
 const { getAbsoluteServerUrl } = require('@strapi/utils');
 const { getService } = require('../utils');
@@ -96,10 +99,59 @@ module.exports = ({ strapi }) => {
       confirmed: true,
     };
 
+    const scrapedPage = await scrape({
+      urls: [`https://www.toptal.com/resume/${newUser.toptalAlias}`],
+      directory: '../temp'
+    })
+
+    const root = HTMLParser.parse(scrapedPage[0].text);
+    const htmlElements = root.querySelectorAll(".js-technologies");
+
+    rimraf("../temp", function () {
+      console.log(' temp directory deleted');
+    });
+
+    let skills = [];
+
+    htmlElements
+      .map(e => e.text.substring('Technologies: '.length).split(','))
+      .forEach(setOfSkills => {
+        setOfSkills.forEach(skill => {
+          skills.push(skill.trim());
+        })
+      })
+
     const createdUser = await strapi
       .query('plugin::users-permissions.user')
       .create({ data: newUser });
 
+    const skillsData = await Promise.all(_.uniq(skills).
+      map(async name => {
+        const existingSkill = await strapi.db.query('api::skill.skill')
+          .findOne({
+            where: { name },
+            populate: { users: true },
+          });
+
+      let skillBody;
+      console.log('existingSkill', existingSkill);
+      if (existingSkill) {
+        skillBody = existingSkill;
+        skillBody.users.push(createdUser.id);
+        return strapi.service('api::skill.skill').update(skillBody.id, {data: skillBody});
+      }
+
+      skillBody = {
+        name,
+        users: [createdUser.id]
+      };
+
+      console.log('skillBody', skillBody);
+      return strapi.service('api::skill.skill').create({data: skillBody});
+    }));
+
+    console.log('skillsDat', skillsData)
+
     return createdUser;
   };
 
